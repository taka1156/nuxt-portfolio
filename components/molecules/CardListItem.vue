<template>
  <div>
    <article class="contents" @click="jump(cardInfo.link ? cardInfo.link : '')">
      <card-title :card-title="cardInfo.title" />
      <figure>
        <card-img :card-img="fomatImg(cardInfo.img.url)" />
        <figcaption>
          <card-text :card-text="cardInfo.content1" />
          <card-text :card-text="cardInfo.content2" />
        </figcaption>
      </figure>
    </article>
  </div>
</template>

<script>
import CardTitle from '../atoms/CardTitle';
import CardImg from '../atoms/CardImg';
import CardText from '../atoms/CardText';

export default {
  name: 'CardItem',
  components: {
    'card-title': CardTitle,
    'card-img': CardImg,
    'card-text': CardText,
  },
  props: {
    cardInfo: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
  methods: {
    fomatImg(img) {
      return `${img}?fit=fillmax&fill-color=gray&w=388&h=312`;
    },
    jump(link) {
      if (link !== '') {
        location.href = link;
      }
    },
  },
};
</script>

<style scoped>
.contents {
  width: 100%;
  height: 100%;
  margin: 2vh auto;
}
</style>
