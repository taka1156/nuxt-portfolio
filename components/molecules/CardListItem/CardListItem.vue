<template>
  <div>
    <article class="card-list-item">
      <figure>
        <base-img-lazy
          :img-url="card.img.url"
          :img-alt="`${card.title}の画像`"
          :size="`${card.link == null ? 'sm' : 'lg'}`"
        />
        <div class="card-list-item__border" />
        <figcaption class="card-list-item__caption">
          <base-title>{{ card.title }}</base-title>
          <base-text>{{ card.content2 }}</base-text>
        </figcaption>
      </figure>
      <base-link v-if="card.link != null" :route-to="card.link" :is-target="true">
        {{ isGithubRepo(card) ? 'GitHub' : 'WebSite' }}
      </base-link>
    </article>
  </div>
</template>

<script>
import BaseTitle from '../../atoms/BaseTitle/BaseTitle';
import BaseText from '../../atoms/BaseText/BaseText';
import BaseLink from '../../atoms/BaseLink/BaseLink';
import BaseImgLazy from '../../atoms/BaseImgLazy/BaseImgLazy';

export default {
  name: 'CardListItem',
  components: {
    'base-title': BaseTitle,
    'base-text': BaseText,
    'base-link': BaseLink,
    'base-img-lazy': BaseImgLazy
  },
  props: {
    /**
     * カードリスト１つ分
     */
    card: {
      type: Object,
      required: true
    }
  },
  methods: {
    isGithubRepo({ link }) {
      /**
       * githubリポジトリかどうかを判定する処理
       */
      return `${link}`.includes('github') && !`${link}`.includes('github.io');
    }
  }
};
</script>

<style scoped>
/* css reset */
figure {
  padding: 0;
  margin: 0;
}

figcaption {
  padding: 0;
  margin: 0;
}

/* css reset */
::v-deep .base-text--extend {
  box-sizing: border-box;
  display: block;
  width: 100%;
  height: 100px;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
}

::v-deep .base-link--extend {
  display: block;
  width: 75%;
  padding: 10px;
  margin: 10px auto;
  font-size: 1em;
  color: gray;
  background-color: white;
  border: 1px solid gray;
  border-radius: 5px;
  outline: none;
}

::v-deep .base-link--extend:active,
::v-deep .base-link--extend:hover,
::v-deep .base-link--extend:focus {
  color: white;
  background-color: gray;
}

::v-deep .base-img--extend {
  padding: 3px;
  margin: 0 auto;
}

.card-list-item {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  border: solid 0.6px rgb(211, 211, 211);
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}

.card-list-item__border {
  border-bottom: 1px dashed black;
}

.card-list-item__caption {
  padding: 10px;
}
</style>
