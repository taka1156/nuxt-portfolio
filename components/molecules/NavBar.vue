<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <div @click="$emit('change-state')">
          <nav-icon :is-open="isOpen" />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import NavIcon from '../atoms/NavIcon';
import NavLogo from '../atoms/NavLogo';

export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
  },
  props: {
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(12, 12, 12, 0.9);
  z-index: 2;
}

.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
