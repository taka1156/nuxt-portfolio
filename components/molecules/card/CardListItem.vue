<template>
  <div>
    <article class="card-list-item">
      <figure>
        <card-img
          :card-img="cardInfo.img.url"
          :img-alt="`${cardInfo.title}の画像`"
          :is-portfolio="cardInfo.link != null"
        />
        <div class="card-list-item__border" />
        <figcaption>
          <card-title>{{ cardInfo.title }}</card-title>
          <base-text class="base-text--card">{{ cardInfo.content2 }}</base-text>
        </figcaption>
      </figure>
      <base-btn v-if="cardInfo.link != null" @click="$emit('jump', cardInfo)">
        {{ isGithubRepo(cardInfo) ? 'GitHub' : 'WebSite' }}
      </base-btn>
    </article>
  </div>
</template>

<script>
import BaseText from '../../atoms/base/BaseText';
import BaseBtn from '../../atoms/base/BaseBtn';
import CardTitle from '../../atoms/card/CardTitle';
import CardImg from '../../atoms/card/CardImg';

export default {
  name: 'CardListItem',
  components: {
    'card-title': CardTitle,
    'card-img': CardImg,
    'base-text': BaseText,
    'base-btn': BaseBtn,
  },
  props: {
    cardInfo: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
  methods: {
    isGithubRepo({ link }) {
      // githubがリンクに含まれてる、かつgithub.ioではないならリポジトリなので表示変更
      return `${link}`.includes('github') && !`${link}`.includes('github.io');
    },
  },
};
</script>

<style scoped>
/* css reset */
figure {
  margin: 0;
  padding: 0;
}

figcaption {
  margin: 0;
  padding: 0;
}

/* css reset */

.card-list-item {
  height: 100%;
  width: 100%;
  margin: 0 auto;
  border: solid 0.6px rgb(211, 211, 211);
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}

.card-list-item__border {
  border-bottom: 1px dashed black;
}
</style>
