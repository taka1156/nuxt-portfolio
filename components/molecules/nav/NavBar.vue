<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <base-btn :use-type="`none`" @btn-click="changeState">
          <nav-icon :is-open="isOpen" />
        </base-btn>
      </div>
    </div>
  </div>
</template>

<script>
import NavLogo from '../../atoms/nav/NavLogo';
import NavIcon from '../../atoms/nav/NavIcon';
import BaseBtn from '../../atoms/base/BaseBtn.vue';
export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
    'base-btn': BaseBtn,
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (開く)
       * @event changeState
       */
      this.$emit('change-state', !this.isOpen);
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(40, 167, 69, 0.9);
  z-index: 2;
}
.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
