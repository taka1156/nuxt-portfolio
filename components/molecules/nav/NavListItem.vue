<template>
  <div>
    <li class="nav-list-item" @click="$emit('change-state')">
      <nuxt-link :to="navItem.to" class="nav-list-item__link">
        <nav-img :nav-img="navItem.img" :img-alt="`${navItem.name}の画像`" />
        <base-text class="base-text--nav">{{ navItem.name }}</base-text>
      </nuxt-link>
    </li>
  </div>
</template>

<script>
import NavImg from '../../atoms/nav/NavImg';
import BaseText from '../../atoms/base/BaseText';

export default {
  name: 'NavListItem',
  components: {
    'nav-img': NavImg,
    'base-text': BaseText,
  },
  props: {
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-list-item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav-list-item__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
