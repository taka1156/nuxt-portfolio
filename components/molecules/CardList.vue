<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-item
        v-for="(cardInfo, index) in cards"
        :key="index"
        :card-info="cardInfo"
        class="card-list__card"
      ></card-item>
    </div>
  </div>
</template>

<script>
import CardListItem from '../molecules/CardListItem';

export default {
  name: 'CardList',
  components: {
    'card-item': CardListItem,
  },
  props: {
    cards: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 27em;
  margin: 2.6em auto;
}
</style>
