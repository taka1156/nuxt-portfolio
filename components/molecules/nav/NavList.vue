<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav-list">
        <ul>
          <nav-item
            v-for="(item, index) in routes"
            :key="index"
            :nav-item="item"
            @change-state="changeState"
          />
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from './NavListItem';
export default {
  name: 'NavList',
  components: {
    'nav-item': NavListItem,
  },
  props: {
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (閉じる)
       * @event changeState
       */
      this.$emit('change-state', false);
    },
  },
};
</script>

<style scoped>
.nav-list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(40, 167, 69, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}
.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}
.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
