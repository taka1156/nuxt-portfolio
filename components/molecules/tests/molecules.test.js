import { mount, RouterLinkStub } from '@vue/test-utils';
import NavBar from '../NavBar.vue';
import NavList from '../NavList.vue';
import NavListItem from '../NavListItem.vue';
import CardList from '../CardList.vue';
import CardListItem from '../CardListItem.vue';

const dummyRouters = [
  { name: 'Top', to: '/home', img: 'http://placehold.jp/150x150.png' },
];

const dummyCards = [
  {
    title: 'ダミータイトル',
    img: { url: 'http://placehold.jp/150x150.png' },
    content2: '詳細',
  },
  {
    title: 'ダミータイトル',
    img: { url: 'http://placehold.jp/150x150.png' },
    content2: '詳細',
  },
];

const dummyCardsLink = [
  {
    title: 'ダミータイトル',
    img: { url: 'http://placehold.jp/150x150.png' },
    content2: '詳細',
    link: 'https://taka1156.site',
  },
  {
    title: 'ダミータイトル',
    img: { url: 'http://placehold.jp/150x150.png' },
    content2: '詳細',
    link: 'https://github.com/taka1156/nuxt-portoflio',
  },
];

// NavBar
describe('NavBar', () => {
  const dummyText = 'ダミーテキスト';
  const navBar = propsData => {
    return mount(NavBar, {
      propsData: {
        logoText: dummyText,
        ...propsData,
      },
    });
  };

  it('NavBar初期値: isOpen, logoText', () => {
    const wrapper = navBar({ isOpen: false });
    // logoText
    expect(wrapper.vm.$options.props.logoText.required).toBe(true);
    expect(wrapper.vm.logoText).toBe(dummyText);
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
  });

  it('logoTextが反映されているか', () => {
    const wrapper = navBar({ isOpen: false });
    const navLogo = wrapper.find('.nav__logo');
    expect(navLogo.text() === dummyText).toBe(true);
  });

  it('ナビゲーションバーを開く', () => {
    const wrapper = navBar({ isOpen: true });
    expect(wrapper.vm.isOpen).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ナビゲーションバーを閉じる', () => {
    const wrapper = navBar({ isOpen: false });
    expect(wrapper.vm.isOpen).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('$emitでメソッド`change-state`が呼ばれた', () => {
    const wrapper = navBar({ isOpen: false });
    wrapper.vm.$emit('change-state');
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});

// NavListItem
describe('NavListItem', () => {
  const navListItem = mount(NavListItem, {
    // <nuxt-link>をRouterLinkStubで探せるようになる
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: {
      navItem: dummyRouters[0],
    },
  });

  it('NavList初期値: navItem', () => {
    // logoText
    expect(navListItem.vm.$options.props.navItem.required).toBe(true);
    console.log(navListItem.vm.$options.props.navItem.default());
    expect(navListItem.vm.navItem).toBe(dummyRouters[0]);
  });

  it('値がDOMに反映されているか', () => {
    const aTag = navListItem.findComponent(RouterLinkStub);
    const imgTag = navListItem.find('img');

    expect(aTag.props().to === dummyRouters[0].to).toBe(true);
    expect(aTag.text() === dummyRouters[0].name).toBe(true);
    expect(imgTag.attributes().src === dummyRouters[0].img).toBe(true);
    // スナップショット
    expect(navListItem.html()).toMatchSnapshot();
  });

  it('$emitでメソッド`change-state`が呼ばれた', () => {
    navListItem.vm.$emit('change-state');
    expect(navListItem.emitted('change-state')).toBeTruthy();
  });
});

// NavList
describe('NavList', () => {
  const navList = propsData => {
    return mount(NavList, {
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      propsData: {
        routers: dummyRouters,
        ...propsData,
      },
    });
  };

  it('NavList初期値: isOpen, routers', () => {
    const wrapper = navList({ isOpen: false });
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
    // routers
    expect(wrapper.vm.$options.props.routers.required).toBe(true);
    console.log(wrapper.vm.$options.props.routers.default());
    expect(wrapper.vm.routers).toBe(dummyRouters);
  });

  it('値がDOMに反映されているか', () => {
    const wrapper = navList({ isOpen: false });
    const aTag = wrapper.findComponent(RouterLinkStub);
    const imgTag = wrapper.find('img');

    expect(aTag.props().to === dummyRouters[0].to).toBe(true);
    expect(aTag.text() === dummyRouters[0].name).toBe(true);
    expect(imgTag.attributes().src === dummyRouters[0].img).toBe(true);
  });

  it('isOpenがtrueの時に`.nav__list`が表示', () => {
    const wrapper = navList({ isOpen: true });
    expect(wrapper.find('.nav__list').element.style.display !== 'none').toBe(true);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('isOpenがfalseの時に`.nav__list`が非表示', () => {
    const wrapper = navList({ isOpen: false });
    expect(wrapper.find('.nav__list').element.style.display === 'none').toBe(true);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('$emitでメソッド`change-state`が呼ばれた', () => {
    const wrapper = navList({ isOpen: true });
    wrapper.vm.$emit('change-state');
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});

// CardListItem
describe('CardListItem', () => {
  const cardListItem = propsData => {
    return mount(CardListItem, {
      propsData: {
        ...propsData,
      },
    });
  };

  it('CardListItemの初期値(linkなし): dummyCards', () => {
    const wrapper = cardListItem({ cardInfo: dummyCards[0] });
    // default
    console.log(wrapper.vm.$options.props.cardInfo.default());
    // const btnTag = wrapper.find('button');
    expect(wrapper.vm.$options.props.cardInfo.required).toBe(true);
    // expect(btnTag == null).toBe(true);
    expect(wrapper.vm.cardInfo).toBe(dummyCards[0]);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('CardListItemの初期値(linkあり): dummyCardsLink', () => {
    const wrapper = cardListItem({ cardInfo: dummyCardsLink[0] });
    const btnTag = wrapper.find('button');
    expect(wrapper.vm.$options.props.cardInfo.required).toBe(true);
    expect(btnTag.text() === 'WebSite').toBe(true);
    expect(wrapper.vm.cardInfo).toBe(dummyCardsLink[0]);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('isGithubRepoメソッド: dummyCardsLink', () => {
    // 空だとエラーを吐くのでデータを入れる
    const wrapper = cardListItem({ cardInfo: dummyCardsLink[0] });
    // githubのリポジトリとでもサイトの時の比較
    expect(wrapper.vm.isGithubRepo(dummyCardsLink[0]) === false).toBe(true);
    expect(wrapper.vm.isGithubRepo(dummyCardsLink[1]) === true).toBe(true);
  });

  it('jumpメソッド', () => {
    // https://remarkablemark.org/blog/2018/11/17/mock-window-location/#update-for-jsdom-14
    delete window.location;
    window.location = { assign: jest.fn() };
    const wrapper = cardListItem({ cardInfo: dummyCardsLink[0] });
    wrapper.vm.jump(dummyCardsLink[0]);
    expect(window.location.assign).toBeCalledWith(dummyCardsLink[0].link);
  });
});

// CardList
describe('CardList', () => {
  const cardList = propsData => {
    return mount(CardList, {
      propsData: {
        ...propsData,
      },
    });
  };

  it('CardListの初期値(linkなし): dummyCards', () => {
    const wrapper = cardList({ cards: dummyCards });
    // default
    console.log(wrapper.vm.$options.props.cards.default());
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(wrapper.vm.cards).toBe(dummyCards);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('CardListの初期値(linkあり): dummyCardsLink', () => {
    const wrapper = cardList({ cards: dummyCardsLink });
    const btnTag = wrapper.findAll('button');
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(btnTag.at(0).text() === 'WebSite').toBe(true);
    expect(btnTag.at(1).text() === 'GitHub').toBe(true);
    expect(wrapper.vm.cards).toBe(dummyCardsLink);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
