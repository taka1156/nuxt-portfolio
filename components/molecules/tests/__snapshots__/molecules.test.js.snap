// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListItem CardListItemの初期値(linkあり): dummyCardsLink 1`] = `
"<div>
  <article class=\\"card-list-item\\">
    <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--lg\\">
      <div class=\\"card-list-item__border\\"></div>
      <figcaption>
        <h2 class=\\"card-title\\">ダミータイトル</h2>
        <p class=\\"base-text base-text--card\\">詳細</p>
      </figcaption>
    </figure> <button class=\\"btn\\">
      WebSite
    </button>
  </article>
</div>"
`;

exports[`CardListItem CardListItemの初期値(linkなし): dummyCards 1`] = `
"<div>
  <article class=\\"card-list-item\\">
    <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--sm\\">
      <div class=\\"card-list-item__border\\"></div>
      <figcaption>
        <h2 class=\\"card-title\\">ダミータイトル</h2>
        <p class=\\"base-text base-text--card\\">詳細</p>
      </figcaption>
    </figure>
    <!---->
  </article>
</div>"
`;

exports[`NavBar ナビゲーションバーを閉じる 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミーテキスト</a>
      <div>
        <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--close\\"></span> <span class=\\"nav-icon__middle\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--close\\"></span></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`NavBar ナビゲーションバーを開く 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミーテキスト</a>
      <div>
        <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--open\\"></span> <span class=\\"nav-icon__middle nav-icon__middle--fade\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--open\\"></span></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`NavList isOpenがfalseの時に\`.nav-list\`が非表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\" style=\\"display: none;\\">
      <ul>
        <div>
          <li class=\\"nav-list-item\\"><a class=\\"nav-list-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav-img\\">
              <p class=\\"base-text base-text--nav\\">Top</p>
            </a></li>
        </div>
      </ul>
    </div>
  </transition-stub>
</div>"
`;

exports[`NavList isOpenがtrueの時に\`.nav-list\`が表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\">
      <ul>
        <div>
          <li class=\\"nav-list-item\\"><a class=\\"nav-list-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav-img\\">
              <p class=\\"base-text base-text--nav\\">Top</p>
            </a></li>
        </div>
      </ul>
    </div>
  </transition-stub>
</div>"
`;

exports[`NavListItem 値がDOMに反映されているか 1`] = `
"<div>
  <li class=\\"nav-list-item\\"><a class=\\"nav-list-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav-img\\">
      <p class=\\"base-text base-text--nav\\">Top</p>
    </a></li>
</div>"
`;
